pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -522,7 +527,8 @@ def make_interaction_term(masked: bool = False) -> qutip.Qobj:
                     return 0 * self.build_operator([("I", "global")])
 
             # make interaction term
-            dipole_interaction = cast(qutip.Qobj, 0)
+            with qutip.CoreOptions(default_dtype="CSR"):
+                dipole_interaction = cast(qutip.Qobj, 0)
```

Comment:
Is this doing anything?

Reply:
Indeed, this is not doing anything, since `cast` is only used to check the typing